src/main/java/com/gildedrose/AgeableRepository.java

```diff
@@ -0,0 +1,82 @@
+package com.gildedrose;
+
+import com.gildedrose.goblins_grotto.Item;
+import com.gildedrose.items.*;
+
+import java.util.*;
+//separate file which is the hashmap construction, this would be imported similar to any database connection
+
+public class AgeableRepository implements DataRepository {
+
+    public Ageable get(String id) {
+        return ItemToAgeableMapper.get(id);
+    }
+
+    public Collection<Ageable> getAll() {
+        return ItemToAgeableMapper.getAll();
+    }
+
+    public void updateAll() {
+        ArrayList<Ageable> allAgeableItems = ItemToAgeableMapper.getAll();
+        for (Ageable ageableItem : allAgeableItems) {
+            ageableItem.age();
+        }
+    }
+}
+
+class ItemToAgeableMapper {
+
+    //This is where other db connections would be established and transformed into an Ageable object
+    static HashMap<String, Item> hashMapDB = populateHashMapDB();
```

Comment:
Question: if you initialized a database connection here in a static field with a method that takes no arguments, how would you test it?

Reply:
That was a problem we ran into 😂 , we are now injecting the the hashmap into the repository pattern so that we can mock it in tests (and we no longer need a mapper as the Datasource (`hashmap<String, Ageable>`) constructed at the top level returns the domain object we would map to if that makes sense)